MaterialLib/TwoPhaseModels/CreateTwoPhaseFlowMaterialProperties.h

```diff
@@ -24,11 +25,12 @@ namespace MaterialLib
 {
 namespace TwoPhaseFlowWithPP
 {
-std::unique_ptr<TwoPhaseFlowWithPPMaterialProperties>
-CreateTwoPhaseFlowMaterialProperties(
+std::tuple<std::unique_ptr<TwoPhaseFlowWithPPMaterialProperties>,
+           BaseLib::ConfigTree>
+createTwoPhaseFlowMaterialProperties(
     BaseLib::ConfigTree const& config,
-    bool const has_material_ids,
-    MeshLib::PropertyVector<int> const& material_ids);
+    boost::optional<MeshLib::PropertyVector<int> const&>
```

Comment:
I forgot if boost::optional handles the const references correctly. Please check if there are any issues with this.